packages/core/echo/echo-schema/src/echo-array.ts

```diff
@@ -116,7 +116,7 @@ export class OrderedSet<T extends DocumentBase> implements Array<T> {
   }
 
   slice(start?: number | undefined, end?: number | undefined): T[] {
-    throw new Error('Method not implemented.');
+    return Array.from(this.values()).slice(start, end);
```

Comment:
Is this working? It seems to just return a sliced JS object -- this doesn't affect that actual model?

Reply:
yes, slice should return a copy https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

Reply:
Add TODOs?